### ISSUE TITLE
Allow to compact (remove redundant transitions) a Transaction Log for achieving purposes

### ISSUE THREAD

**Issue by Konard** [Konard/LinksPlatform#47]

**Comment by werkzeuge** [comment]
Типа ПРОЛОГ-системы, решающие правила? Цель, правила вывода?

**Comment by Konard** [comment]
Нет. Здесь имеется виду, что существуют изменения, которые отменяют друг друга, например мы создали связь, потом её удалил, и так 20 раз, но по факту, в логе можно оставить только конечное состояние этой связи, либо она есть, либо её нет. Однако в отличии от упаковки самой бд, в логе важно оставить информацию о том, когда было выполнено последнее действие (т.е. в логе дополнительно хранится отметка времени).

**Comment by werkzeuge** [comment]
Компактификация данных (дефрагментация или укрупнение знаний).

**Comment by Konard** [comment]
Компактификация данных за счёт удаления избыточного объёма знаний. Что имеется ввиду под дефрагментацией?

**Comment by werkzeuge** [comment]
Дефрагментация - уменьшение физического объема хранения данных при сохранении содержания. "Складывание в кучки", говоря обыденным языком.

**Comment by Konard** [comment]
Физический объём хранения данных не обязательно уменьшается при таком процессе. И не всегда такое кучкование имеет смысл.

**Comment by werkzeuge** [comment]
При компактификации не обязательно уменьшается физический объем? Пример можешь привести?

**Comment by Konard** [comment]
> Физический объём хранения данных не обязательно уменьшается при таком процессе. И не всегда такое кучкование имеет смысл.
